client/src/components/video.tsx

```diff
@@ -1,327 +1,327 @@
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import React, { useState } from "react";
+import ReactPlayer from "react-player";
+import { useSelector, useDispatch } from "react-redux";
+import { Star, StarBorder } from "@material-ui/icons";
+import InfoOutlinedIcon from "@material-ui/icons/InfoOutlined";
+import { videoUrl, subtitleUrl, idleUrl } from "api";
+import LoadingSpinner from "components/video-spinner";
+import MessageStatus from "components/video-status";
+import { chromeVersion } from "utils";
+import {
+  answerFinished,
+  faveMentor,
+  mentorAnswerPlaybackStarted,
+} from "store/actions";
+import { State } from "types";
+import "styles/layout.css";
+
+const subtitlesSupported = Boolean(!chromeVersion() || chromeVersion() >= 62);
+
+interface VideoData {
+  src: string;
+  subtitles: string;
+}
+
+function Video(args: { playing?: boolean }): JSX.Element {
+  const { playing = false } = args;
+  const dispatch = useDispatch();
+  const curMentor = useSelector<State, string>((state) => state.curMentor);
+  const video = useSelector<State, VideoData | null>((state) => {
+    if (!state.curMentor) {
+      return null;
+    }
+    const m = state.mentorsById[state.curMentor];
+    if (!m) {
+      return null;
+    }
+    return {
+      src: state.isIdle ? idleUrl(m.mentor) : videoUrl(m.answer_media || []),
+      subtitles:
+        subtitlesSupported && !state.isIdle
+          ? subtitleUrl(m.answer_media || [])
+          : "",
+    };
+  });
+
+  interface HeaderMentorData {
+    _id: string;
+    name: string;
+    title: string;
+  }
+
+  const mentorName = useSelector<State, HeaderMentorData | null>((state) => {
+    if (!state.curMentor) {
+      return null;
+    }
+    const m = state.mentorsById[state.curMentor];
+    if (!(m && m.mentor)) {
+      return null;
+    }
+    return {
+      _id: m.mentor._id,
+      name: m.mentor.name,
+      title: m.mentor.title,
+    };
+  });
+
+  const [hideLinkLabel, setHideLinkLabel] = useState<boolean>(false);
+  const isIdle = useSelector<State, boolean>((state) => state.isIdle);
+
+  const lastAnswerLink = useSelector<State, string>((state) => {
+    const totalMentors = Object.keys(state.mentorsById).length;
+    const chatAnswers = state.chat.messages.map((m) => {
+      return !m.isUser ? findLinks(m.text) : "";
+    });
+    const lastMentorAnswers = chatAnswers.slice(-totalMentors);
+    return getLastAnswerLink(lastMentorAnswers.reverse());
+  });
+
+  function getLastAnswerLink(lastMentorAnswers: Array<string>): string {
+    for (let i = 0; i < lastMentorAnswers.length; i++) {
+      if (lastMentorAnswers[i] !== "") {
+        return lastMentorAnswers[i];
+      }
+    }
+    return "";
+  }
+
+  function findLinks(text: string): string {

```

Comment:
do we not already have links parsed out in the chat data in redux? If we're only parsing ASK links there...let's move this over and parse the `WebLink` there as well

Reply:
Video web-links parsing moved to the reducer: https://github.com/mentorpal/mentor-client/pull/87/commits/8fd9350821683d0ac8ef7007395c6382daf72a5d

New two tests added: https://github.com/mentorpal/mentor-client/pull/87/commits/6d37f1e51879969ebdc0f6174343ae6e1f22ae71